pkg/query-service/pqlEngine/engine.go

```diff
@@ -42,22 +44,37 @@ func NewPqlEngine(config *pconfig.Config) (*PqlEngine, error) {
 
 	logLevel := plog.AllowedLevel{}
 	logLevel.Set("debug")
-	logger := plog.New(logLevel)
+
+	allowedFormat := promlog.AllowedFormat{}
+	allowedFormat.Set("logfmt")
+
+	promlogConfig := promlog.Config{
+		Level:  &logLevel,
+		Format: &allowedFormat,
+	}
+
+	logger := plog.New(&promlogConfig)
 
 	opts := pql.EngineOpts{
-		Logger:        log.With(logger, "component", "promql evaluator"),
-		Reg:           nil,
-		MaxConcurrent: 20,
-		MaxSamples:    50000000,
-		Timeout:       time.Duration(2 * time.Minute),
+		Logger:     log.With(logger, "component", "promql evaluator"),
+		Reg:        nil,
+		MaxSamples: 50000000,
+		Timeout:    time.Duration(2 * time.Minute),
 	}
 
 	e := pql.NewEngine(opts)
 	startTime := func() (int64, error) {
 		return int64(pmodel.Latest), nil
 	}
 
-	remoteStorage := premote.NewStorage(log.With(logger, "component", "remote"), startTime, time.Duration(1*time.Minute))
+	remoteStorage := premote.NewStorage(
+		log.With(logger, "component", "remote"),
+		nil,
```

Comment:
please recheck the use of `nil` in `prometheus.Registerer`. I don't have much context here.